### ISSUE TITLE
How to communicate with other ct

### ISSUE THREAD

**Issue by YYRise** [google/certificate-transparency-go#644]
When the logID of the certificate is not under the current ct tree, how to query or verify the certificate?

**Comment by RJPercival** [comment]
Could you provide more detail, e.g. a step-by-step example of what you're trying to do please?

**Comment by YYRise** [comment]
> Could you provide more detail, e.g. a step-by-step example of what you're trying to do please?

Thanks for the response。Certificate A was signed by another ct, not my ct. How can I read the signature information of certificate A through my ct。Can you understand this?

**Comment by RJPercival** [comment]
Are you talking about verifying the signatures in a certificate's SCTs? Is your problem that they're signed by CT logs for which you don't have the public keys?

**Comment by YYRise** [comment]
> Are you talking about verifying the signatures in a certificate's SCTs? Is your problem that they're signed by CT logs for which you don't have the public keys?

I'm sorry for not expressing the distress you have caused. Currently I have deployed the ct_server、trillian_log_server and trillian_log_signer, and the certificate signing information is stored in my mysql. The user wants to access the certificate that is not in my mysql through the service I deployed. What should I do? If the certificate is signed by you, how can I get it in my service. ct_server started with my own tree root ID, but the certificate is in your tree.

**Comment by RJPercival** [comment]
Ah ok, let me try again to see if I understand correctly. You're talking about a certificate being submitted to another CT log (e.g. Google's), but a user wanting to query your CT log for that certificate (which doesn't contain it)? You want your CT log to contain all of the certificates found in any other log?

**Comment by YYRise** [comment]
> Ah ok, let me try again to see if I understand correctly. You're talking about a certificate being submitted to another CT log (e.g. Google's), but a user wanting to query your CT log for that certificate (which doesn't contain it)? You want your CT log to contain all of the certificates found in any other log?

Yes, that's what I meant. Thank you for understanding

**Comment by RJPercival** [comment]
You can copy all of the certificates from one CT log to another using the following command. It will take many days to copy a large CT log.
```shell
go run github.com/google/certificate-transparency-go/preload/preloader \
  --source_log_uri "https://ct.googleapis.com/rocketeer"
  --target_log_uri "https://your-log-url-here"
```

Why can't your users query those other logs to find the certificate though? We publish [a log list](http://www.gstatic.com/ct/log_list/v2/log_list.json) that helps users map log IDs to URLs for this reason (amongst others).

**Comment by YYRise** [comment]
> You can copy all of the certificates from one CT log to another using the following command. It will take many days to copy a large CT log.
> 
> ```shell
> go run github.com/google/certificate-transparency-go/preload/preloader \
>   --source_log_uri "https://ct.googleapis.com/rocketeer"
>   --target_log_uri "https://your-log-url-here"
> ```
> 
> Why can't your users query those other logs to find the certificate though? We publish [a log list](http://www.gstatic.com/ct/log_list/v2/log_list.json) that helps users map log IDs to URLs for this reason (amongst others).

Thanks, very much! Very helpful for me, I need to learn these, if I have questions, I will bother you again.

**Comment by RJPercival** [comment]
Happy to help!
